nemo/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -272,26 +273,11 @@ def on_train_end(self, trainer, pl_module):
                     logging.debug(f'Last checkpoint {self.last_model_path} already saved')
                 else:
                     super()._save_last_checkpoint(trainer, monitor_candidates)
+            if self.save_context_on_train_end and not self.always_save_context and is_global_rank_zero():
+                TrainerContext.from_trainer(trainer).io_dump(ckpt_to_dir(self.last_model_path) / "artifacts")
```

Comment:
I would prefer a folder called `context`. In there we could have a folder artifacts where we can put things like tokenizer etc.